pkg/volume/storageos/storageos_util.go

```diff
@@ -0,0 +1,433 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storageos
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/volume"
+
+	"github.com/golang/glog"
+	storageosapi "github.com/storageos/go-api"
+	storageostypes "github.com/storageos/go-api/types"
+)
+
+const (
+	losetupPath   = "losetup"
+	apiSecretName = "storageos-api"
+
+	modeBlock deviceType = iota
+	modeFile
+	modeUnsupported
+
+	ErrDeviceNotFound     = "device not found"
+	ErrDeviceNotSupported = "device not supported"
+	ErrNotAvailable       = "not available"
+)
+
+type deviceType int
+
+// storageosVolume describes a provisioned volume
+type storageosVolume struct {
+	ID          string
+	Name        string
+	Namespace   string
```

Comment:
does this correspond to the name of a kubernetes namespace? document it either way, and possibly rename if this is a StorageOS namespace